### PR TITLE
anim_goal_push_hit_by_weapon and fixes

### PR DESCRIPTION
**Pull request by anatoliy-savchak**
* added anim_goal_push_hit_by_weapon;
* LegacyCombatSystem::UseItem fixed when many items in NPC inventory;
* fixed PyDice.packed to work with long and int;
* fixed  ItemCallbacks::UseableItemActionCheck to allow unidentified usage for non party NPC;

### CODE REVIEW

File: TemplePlus/animgoals/anim.cpp [hunk 1]
```diff
@@ -2437,3 +2408,39 @@ float AnimSystemHooks::HookedGetRunSpeed(objHndl handle, obj_f field){ // this i
...
+    agd.scratchVal6.number = 5;
+    AnimSlotId* animIdGlobal = addresses.animIdGlobal;
+    if (agd.Push(animIdGlobal))
+        return TRUE;
+
+    return FALSE;
+}
+
+int AnimSystem::PushGoalHitByWeapon(objHndl attacker, objHndl defender) {
+    return animHooks.PushGoalHitByWeapon(attacker, defender);
```
Comment (DudeMcDude) [inline TemplePlus/animgoals/anim.cpp:2445]: It should be the other way around, hooks calling from AnimGoals...

File: TemplePlus/condition.cpp [hunk 2]
```diff
@@ -5249,7 +5249,8 @@ int ItemCallbacks::UseableItemActionCheck(DispatcherCallbackArgs args){
 	int useMagicDeviceSkillBase = critterSys.SkillBaseGet(args.objHndCaller, skill_use_magic_device);
 
 	// ensure is identified
-	if (objType != obj_t_food && !inventory.IsIdentified(itemHandle)){
+	if (objType != obj_t_food && !inventory.IsIdentified(itemHandle) && party.IsInParty(args.objHndCaller)){
+		logger->debug("Item is not identified! Item: {}", description.getDisplayName(itemHandle));
```
Comment (DudeMcDude) [inline TemplePlus/condition.cpp:5253]: ObjHndl has a stream operator now, no need for manually doing getDisplayName.

File: TemplePlus/python/python_object.cpp [hunk 3]
```diff
@@ -2900,10 +2913,13 @@ static PyObject* PyObjHandle_SpellKnownAdd(PyObject* obj, PyObject* args) {
 	int spellIdx;
 	int spellClassCode;
 	int slotLevel;
-	if (!PyArg_ParseTuple(args, "iii:objhndl.spell_known_add", &spellIdx, &spellClassCode, &slotLevel)) {
+	int isDomain = 0;
+	if (!PyArg_ParseTuple(args, "iii|i:objhndl.spell_known_add", &spellIdx, &spellClassCode, &slotLevel, &isDomain)) {
 		return 0;
 	}
-	spellSys.SpellKnownAdd(self->handle, spellIdx, (spellClassCode & 0x7F) | 0x80, slotLevel, 1, 0);
+	if (!isDomain)
```
Comment (DudeMcDude) [inline TemplePlus/python/python_object.cpp:2920]: For consistency, I'd  like to enforce the bitmask on the is_domain case too.

File: TemplePlus/python/python_object.cpp [hunk 4]
```diff
@@ -2915,10 +2931,13 @@ static PyObject* PyObjHandle_SpellMemorizedAdd(PyObject* obj, PyObject* args) {
...
 	int spellClassCode;
 	int slotLevel;
-	if (!PyArg_ParseTuple(args, "iii:objhndl.spell_memorized_add", &spellIdx, &spellClassCode, &slotLevel)) {
+	int isDomain = 0;
+	if (!PyArg_ParseTuple(args, "iii|i:objhndl.spell_memorized_add", &spellIdx, &spellClassCode, &slotLevel, &isDomain)) {
 		return 0;
 	}
-	spellSys.SpellMemorizedAdd(self->handle, spellIdx, (spellClassCode & 0x7F) | 0x80, slotLevel, 2, 0);
+	if (!isDomain)
+		spellClassCode = (spellClassCode & 0x7F) | 0x80;
```
Comment (DudeMcDude) [inline TemplePlus/python/python_object.cpp:2939]: Same here

**Review by DudeMcDude — CHANGES_REQUESTED** [review]
Added remarks inside code

**Comment by anatoliy-savchak** [comment]
Updated. @DudeMcDude please confirm

**Pull request merged**